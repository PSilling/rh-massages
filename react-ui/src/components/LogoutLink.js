// react imports
import React, { Component } from 'react';

// util imports
import Auth from '../utils/Auth.js';
import _t from '../utils/Translations.js';
import Util from '../utils/Util.js';

/**
 * Link that redirects to Keycloak logout and the application after backend logout.
 */
class LogoutLink extends Component {

  logout = () => {
      fetch(Util.LOGOUT_URL, {
        method: 'get',
        headers: {
          'Authorization': 'Bearer ' + Auth.getToken()
        }
      }).then(function(response) {
        if (response.ok) {
          Auth.keycloak.logout();
        } else {
          Util.notify("error", _t.translate('Your request has ended unsuccessfully.'),
            _t.translate('An error occured!'));
        }
      });
    }

  render() {
    return(
      <button type="button" className="btn btn-link navbar-btn" onClick={() => this.logout()}>
        <span className="glyphicon glyphicon-log-out"></span>&nbsp;
          { _t.translate('Logout') }
      </button>
    )
  }
}

export default LogoutLink
