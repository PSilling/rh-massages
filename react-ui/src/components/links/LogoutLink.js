// react imports
import React, { Component } from "react";

// module imports
import { NavLink } from "reactstrap";

// util imports
import Auth from "../../util/Auth";
import _t from "../../util/Translations";
import Util from "../../util/Util";

/**
 * Link that redirects to Keycloak logout and the application after server logout.
 */
class LogoutLink extends Component {
  logout = () => {
    fetch(Util.LOGOUT_URL, {
      method: "get",
      headers: {
        Authorization: `Bearer ${Auth.getToken()}`
      }
    }).then(response => {
      if (response.ok) {
        Auth.keycloak.logout();
      } else {
        Util.notify(
          "error",
          _t.translate("Your request has ended unsuccessfully."),
          _t.translate("An error occurred!")
        );
      }
    });
  };

  render() {
    return (
      <NavLink onClick={this.logout} style={{ cursor: "pointer" }}>
        <span className="fas fa-sign-out-alt" />
        &nbsp;
        {_t.translate("Logout")}
      </NavLink>
    );
  }
}

export default LogoutLink;
