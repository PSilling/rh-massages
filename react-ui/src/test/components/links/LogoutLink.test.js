// react imports
import React from 'react';
import TestRenderer from 'react-test-renderer';

// test imports
import LogoutLink from '../../../components/links/LogoutLink';

test('renders content correctly', () => {
  const testRenderer = TestRenderer.create(<LogoutLink />),
        testInstance = testRenderer.root;

  let button = testInstance.findByType('button'),
      icon = testInstance.findByProps({ className: "glyphicon glyphicon-log-out" }),
      treeJSON = testRenderer.toJSON();

  testInstance.instance.logout = jest.fn();
  testRenderer.update(<LogoutLink />);
  button.props.onClick();

  expect(testInstance.instance.logout).toHaveBeenCalledTimes(1);
  expect(treeJSON).toMatchSnapshot();
});
