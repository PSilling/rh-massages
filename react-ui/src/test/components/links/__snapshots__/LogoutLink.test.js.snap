// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content correctly 1`] = `
<button
  className="btn btn-link navbar-btn"
  onClick={[Function]}
  type="button"
>
  <span
    className="glyphicon glyphicon-log-out"
  />
   
  Odhlásit
</button>
`;
